internal/component/database_observability/mysql/collector/explain_plan.go

```diff
@@ -0,0 +1,750 @@
+	explainPlanOutputOperationHashJoin             explainPlanOutputOperation = "Hash Join"
+	explainPlanOutputOperationMergeJoin            explainPlanOutputOperation = "Merge Join"
+	explainPlanOutputOperationGroupingOperation    explainPlanOutputOperation = "Grouping Operation"
+	explainPlanOutputOperationOrderingOperation    explainPlanOutputOperation = "Ordering Operation"
+	explainPlanOutputOperationDuplicatesRemoval    explainPlanOutputOperation = "Duplicates Removal"
+	explainPlanOutputOperationMaterializedSubquery explainPlanOutputOperation = "Materialized Subquery"
+	explainPlanOutputOperationAttachedSubquery     explainPlanOutputOperation = "Attached Subquery"
+	explainPlanOutputOperationUnion                explainPlanOutputOperation = "Union"
+)
+
+type explainPlanAccessType string
+
+const (
+	explainPlanAccessTypeAll   explainPlanAccessType = "all"
+	explainPlanAccessTypeIndex explainPlanAccessType = "index"
+	explainPlanAccessTypeRange explainPlanAccessType = "range"
+	explainPlanAccessTypeRef   explainPlanAccessType = "ref"
+	explainPlanAccessTypeEqRef explainPlanAccessType = "eq_ref"
+)
+
+type explainPlanJoinAlgorithm string
+
+const (
+	explainPlanJoinAlgorithmHash       explainPlanJoinAlgorithm = "hash"
+	explainPlanJoinAlgorithmMerge      explainPlanJoinAlgorithm = "merge"
+	explainPlanJoinAlgorithmNestedLoop explainPlanJoinAlgorithm = "nested_loop"
+)
+
+type explainPlanOutput struct {
+	Metadata metadataInfo `json:"metadata"`
+	Plan     planNode     `json:"plan"`
+}
+
+type metadataInfo struct {
+	DatabaseEngine  string `json:"database_engine"`
+	DatabaseVersion string `json:"database_version"`
+	QueryIdentifier string `json:"query_identifier"`
+	GeneratedAt     string `json:"generated_at"`
+}
+
+type planNode struct {
+	Operation explainPlanOutputOperation `json:"operation"`
+	Details   nodeDetails                `json:"details"`
+	Children  []planNode                 `json:"children,omitempty"`
+}
+
+type nodeDetails struct {
+	EstimatedRows int64                     `json:"estimatedRows"`
+	EstimatedCost *float64                  `json:"estimatedCost,omitempty"`
+	TableName     *string                   `json:"tableName,omitempty"`
+	Alias         *string                   `json:"alias,omitempty"`
+	AccessType    *explainPlanAccessType    `json:"accessType,omitempty"`
+	KeyUsed       *string                   `json:"keyUsed,omitempty"`
+	JoinType      *string                   `json:"joinType,omitempty"`
+	JoinAlgorithm *explainPlanJoinAlgorithm `json:"joinAlgorithm,omitempty"`
+	Condition     *string                   `json:"condition,omitempty"`
+	GroupByKeys   []string                  `json:"groupByKeys,omitempty"`
+	SortKeys      []string                  `json:"sortKeys,omitempty"`
+	Warning       *string                   `json:"warning,omitempty"`
+}
+
+func NewExplainPlanOutput(logger log.Logger, dbVersion string, digest string, explainJson []byte, generatedAt string) (*explainPlanOutput, error) {
```

Comment:
Can we make this package private?

Reply:
Oops, thought I had in the previous batch.. Done.